### PR TITLE
Add view mode

### PR DESCRIPTION
**Pull request by pentacular**

### CODE REVIEW

File: es6/jsxcad-ui-app.js [hunk 1]
```diff
@@ -38697,89 +38696,162 @@ class JsEditorUi extends ReactDOM$2.PureComponent {
...
+            const line = initSourceLocation.start.line;
+            orderedNotes.push({
+              domElement,
+              notes,
+              line
+            });
           }
 
-          break;
+          orderedNotes.sort((a, b) => a.line - b.line); // FIX: This produces flicker.
```
Comment (BarbourSmith) [inline es6/jsxcad-ui-app.js:38798]: I didn't run into the flicker which is nice :)

File: es6/jsxcad-ui-app.js [hunk 2]
```diff
@@ -39809,7 +39881,9 @@ class App extends ReactDOM$2.Component {
 
               if (domElementByHash.has(entry.hash)) {
                 // Reuse the element we built earlier
-                console.log(`Re-appending ${entry.hash} to ${path}/${id}`);
+                console.log(`Re-appending ${entry.hash} to ${path}/${id}`); // FIX: This is a problem because it removes the child from where it
```
Comment (BarbourSmith) [inline es6/jsxcad-ui-app.js:39884]: I did have an issue where switching from view to edit mode would require re-running the notebook for the figures to appear sometimes

File: es6/jsxcad-ui-app.js [hunk 3]
```diff
@@ -41345,7 +41478,75 @@ class App extends ReactDOM$2.Component {
...
+    const SHIFT = 16;
+    const key = e.which || e.keyCode || 0;
+
+    switch (key) {
+      case CONTROL:
+      case SHIFT:
+        return true;
+    }
+
+    const {
```
Comment (BarbourSmith) [inline es6/jsxcad-ui-app.js:41508]: Having a UI button to toggle the mode would be awesome in addition to ctrl+E

**Review by BarbourSmith — APPROVED** [review]
Looks great to me!

**Comment by pentacular** [comment]
> cool. let me know when it is live :)

Actually, it's live now on alpha/

**Pull request merged**